cmd/kubeconfig/create.go

```diff
@@ -24,42 +25,79 @@ import (
 // TODO: NEXT: incorporate into an fzf workflow
 
 const description string = `
-This command renders a kubeconfig with a context for every HostedCluster resource.
-The contexts are named based on the HostedCluster following the pattern:
+This command renders kubeconfigs for HostedClusters.
+
+If a single cluster is identified, the kubeconfig is printed to stdout.
+
+If all clusters are selected, this command renders a kubeconfig with a context
+for every HostedCluster resource. The contexts are named based on the
+HostedCluster following the pattern:
 
     {hostedcluster.namespace}-{hostedcluster.name}
 
 The kubeconfig for each cluster is based on the secret referenced by the status
 of the HostedCluster itself.
 `
 
-// NewCreateCommand returns a command which creates a combined kubeconfig
-// from HostedCluster resources.
+type Options struct {
+	All       bool
+	Namespace string
+	Name      string
+}
+
+func (o Options) Validate() error {
+	if o.All && (len(o.Namespace) > 0 || len(o.Name) > 0) {
+		return fmt.Errorf("either all hostedclusters or a single one must be specified, not both")
+	}
+	if !o.All && (len(o.Namespace) == 0 || len(o.Name) == 0) {
+		return fmt.Errorf("both namespace and name must be specified")
+	}
+	return nil
+}
+
+// NewCreateCommand returns a command which can render kubeconfigs for HostedCluster
+// resources.
 func NewCreateCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:          "kubeconfig",
-		Short:        "Creates a combined kubeconfig from hostedcluster resources",
+		Short:        "Renders kubeconfigs for hostedcluster resources.",
 		Long:         description,
 		SilenceUsage: true,
 	}
 
-	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+	opts := Options{
+		All: false,
```

Comment:
In the current behavior, doesn't `All` default to `true`?

Reply:
Removed the `--all-` flag and made the previous behavior implicit when no cluster is addressed